lib/api/environment.rb

```diff
@@ -1,7 +1,30 @@
 module Api
   class Environment
     def self.normalized_attributes
-      @normalized_attributes ||= {:time => {}, :url => {}, :resource => {}, :encrypted => {}}
+      @normalized_attributes ||= {
+        :time      => time_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = true },
+        :url       => {"href" => true},
+        :resource  => {"image_href" => true},
+        :encrypted => encrypted_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = true }
+      }
+    end
+
+    def self.encrypted_attributes
+      @encrypted_attributes ||= %w(password) |
+                                ::MiqRequestWorkflow.all_encrypted_options_fields.map(&:to_s) |
+                                ::Vmdb::Settings::PASSWORD_FIELDS.map(&:to_s)
+    end
+
+    def self.time_attributes
+      @time_attributes ||= begin
```

Comment:
The begin/end is redundant here.  Also I'm not a fan of the mega indent, personally...I prefer

```ruby
   def self.time_attributes
     @time_attributes ||=
       ApiConfig.collection... do
         
       end
   end